moduleit/repl-environment.nix

```diff
@@ -0,0 +1,511 @@
+{ config, lib, pkgs, ... }:
```

Comment:
i feel like `repl-environment` isn't a great name for this. Neither `repl` or `environment` represent what this is. I'd call it `moduleDeclaration` or something closer to what this is.

Reply:
Yeah, I think it makes more sense to move it to the overlay and have goval reference it from there.